Add option to auto collapse views

santhoshkumarCodes opened the pull request:
Fixes: #34470 Add option to auto-collapse Views section
<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

## In this PR:

### Frontend:
- Made the entire **left sidebar scrollable**.
- Added a new option in the `...` menu for **auto-collapsing views** when scrolled down.
- **Added box-shadow** to the sidebar on scroll for visual feedback.
- Improved UI by making the `...` menu **visible only on hover**.
- Added a new **preference setting** for auto-collapsing views.
- Made the frontend **respect the user preference** and **removed localStorage logic** initially used for state tracking.
- Added node test for `user_settings` event handling in `server_events_dispatch.js`.
- **Fixed all known bugs** related to this feature.
- **Manually tested** to ensure correct behavior across different scenarios.

### Backend:
- Added a new boolean field: `auto_collapse_views`, including **schema migration**.
- Updated `property_types` to support the new setting.
- Added **automated test coverage** for the new preference.
- Documented the new boolean field: `auto_collapse_views`.

**Screenshots and screen captures:**
![Peek 2025-05-08 17-45](https://github.com/user-attachments/assets/afdc8e02-9116-4244-a711-4e4f5e2cdcf4)
![Screenshot from 2025-05-09 01-41-18](https://github.com/user-attachments/assets/bef6d5c6-9fdb-4cc6-9c0f-cd2fbde3c083)
![Screenshot from 2025-05-09 01-43-13](https://github.com/user-attachments/assets/ee907359-0848-4dfd-b783-5a463784d23e)
![Screenshot from 2025-05-09 01-43-35](https://github.com/user-attachments/assets/6c45e7e0-56c7-414c-8841-5c295c5cdf58)
![Screenshot from 2025-05-09 01-44-00](https://github.com/user-attachments/assets/8e0617a2-fda7-43fd-a82a-bec5bcb3d430)
![Screenshot from 2025-05-09 01-45-01](https://github.com/user-attachments/assets/8e81b430-582e-4ffe-8b14-cde1b28d1e51)
![Screenshot from 2025-05-09 01-45-31](https://github.com/user-attachments/assets/1f043024-0232-4f36-a9ad-321b06697b10)
![Screenshot from 2025-05-09 01-45-51](https://github.com/user-attachments/assets/aec54847-29be-4a91-9b48-e168006374a3)


<details>

<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [x] Highlights technical choices and bugs encountered.
- [x] Calls out remaining decisions and concerns.
- [x] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [x] Strings and tooltips.
- [x] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.
</details>

santhoshkumarCodes commented:
@alya [Previous PR](https://github.com/zulip/zulip/pull/34582) got messed up (Never thought dropping all the commits would close the PR), so I made a new PR with a cleaner commit history.

alya commented:
Please do another round of careful manual testing. In the first 30 seconds of playing with the PR, I saw:

1. Extraneous underlines when hovering over views.

![Screenshot 2025-05-17 at 15 04 09@2x](https://github.com/user-attachments/assets/1a62fc04-effc-4378-93a0-4e8234827fe7)

2. An extraneous blank tooltip when hovering over the new menu option.


![image](https://github.com/user-attachments/assets/a2a89f20-4a10-4271-ba2e-8a3a5ac76dd9)

santhoshkumarCodes commented:
@alya I believe the underline behavior on hover also exists in the current main codebase and wasn’t introduced by this PR — but I’ve fixed it now anyway:

![Screenshot from 2025-05-18 12-55-41](https://github.com/user-attachments/assets/97486b7d-5174-486b-8789-1ca58ad7c2e0)

Apologies for the blank tooltip — I had experimented with it and forgot to remove the `tippy-left-sidebar-tooltip` class. That’s cleaned up now.

I've also done a full review of the PR and fixed a few typos along the way.

alya commented:
The hover underlines are not present in `main` when I test, so I don't understand how there could be a bug unrelated to this PR.

santhoshkumarCodes commented:
@alya Thanks! I want to clarify that my PR doesn't modify any draft-related files (`drafts.ts`), and I'm unable to reproduce this error locally. Does this not suggest that the error might be unrelated to my changes?

Could you help me understand:

Does this error also occur on the main branch without my changes?
Are there specific steps to reproduce it?

alya commented:
> as the hover underlines are not present in the main

Can you explain why you said earlier that they are present in `main`?

santhoshkumarCodes commented:
> I got the error on your PR, but not in `main`, immediately upon opening the app. You can ask in #development help in the development community to see if folks can help you figure out what's up.

Thanks for letting me know! I’ll work with #development help to debug this.

santhoshkumarCodes commented:
> Can you explain why you said earlier that they are present in `main`?

When comparing the diff between `main` and `my-branch`, the `.left-sidebar-navigation-label-container` has `text-decoration: none` only on focus. I changed it to always have `text-decoration: none`, so I initially thought it was already present in `main`. Since you've confirmed that it's not, I assumed there might be something else causing this issue.

alya commented:
I see. In the future, I recommend testing things directly, rather than making assumptions based on the code.

santhoshkumarCodes commented:
Sure, I will test things directly!

santhoshkumarCodes commented:
@alya Now everything is resolved.
![Screenshot from 2025-05-31 14-26-33](https://github.com/user-attachments/assets/aab189fe-64ab-4a2f-a3f4-70216e0b929f)
![Screenshot from 2025-05-31 14-26-47](https://github.com/user-attachments/assets/7621dfc8-1466-4f00-a2a6-535f1dd171b8)

alya commented:
I'm still seeing extraneous underlines on the view buttons when views are collapsed, and there's awkward vertical motion on the headings when going between collapsed/expanded.

Please be mindful that in order for us to be able to continue reviewing your work, you will need to review it *carefully* yourself before asking for feedback from others.
